odoo/tools/misc.py

```diff
@@ -1608,13 +1608,16 @@ def format_decimalized_amount(amount: float, currency=None) -> str:
     return "%s %s" % (formated_amount, currency.symbol or '')
 
 
-def format_amount(env: Environment, amount: float, currency, lang_code: str | None = None) -> str:
+def format_amount(env: Environment, amount: float, currency, lang_code: str | None = None, trailing_zeroes: bool = True) -> str:
     fmt = "%.{0}f".format(currency.decimal_places)
     lang = env['res.lang'].browse(get_lang(env, lang_code).id)
 
     formatted_amount = lang.format(fmt, currency.round(amount), grouping=True)\
         .replace(r' ', u'\N{NO-BREAK SPACE}').replace(r'-', u'-\N{ZERO WIDTH NO-BREAK SPACE}')
 
+    if not trailing_zeroes:
+        formatted_amount = re.sub(r'\.?0+$', '', formatted_amount)
```

Comment:
Unfortunately we can do formatting before amount is formatted in res_lang, I have made modification in RE so that it takes the language `decimal_point`. I have added a test regarding this(a language with same decimal_point and thousands_sep). You can have a look or else we could do something like this
 ```
fmt = "%.{0}f".format(0 if not trailing_zeroes and amount.is_interger() else currency.decimal_places)
 ```